packages/across-monitor/src/AcrossMonitor.ts

```diff
@@ -0,0 +1,151 @@
+import Web3 from "web3";
+const { toBN, toWei } = Web3.utils;
+
+import winston from "winston";
+
+import { createEtherscanLinkMarkdown, createFormatFunction, PublicNetworks } from "@uma/common";
+
+import { RelayEventProcessor } from "./RelayEventProcessor";
+
+import type { BridgePoolData, InsuredBridgeL1Client } from "@uma/financial-templates-lib";
+import type { AcrossMonitorConfig } from "./AcrossMonitorConfig";
+import type { EventInfo } from "./RelayEventProcessor";
+
+export class AcrossMonitor {
+  // Discovered bridge pools are populated after update().
+  public bridgePools: { [key: string]: BridgePoolData } = {};
+
+  // Block range to search is only defined on calling update().
+  private startingBlock: number | undefined = undefined;
+  private endingBlock: number | undefined = undefined;
+
+  // relayEventProcessor Module used to fetch and process relay events.
+  private relayEventProcessor: RelayEventProcessor;
+
+  /**
+   * @notice Constructs new AcrossMonitor Instance.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} monitorConfig Across monitor configuration parameters.
+   * @param {Object} l1Client InsuredBridgeL1Client used for bridge pool discovery.
+   */
+  constructor(
+    readonly logger: winston.Logger,
+    readonly monitorConfig: AcrossMonitorConfig,
+    readonly l1Client: InsuredBridgeL1Client
+  ) {
+    this.relayEventProcessor = new RelayEventProcessor();
+  }
+
+  async update(): Promise<void> {
+    // Update l1Client for bridge pool discovery.
+    await this.l1Client.update();
+    this.bridgePools = this.l1Client.bridgePools;
+    this.relayEventProcessor.bridgePools = this.l1Client.bridgePools;
+
+    // In serverless mode (pollingDelay === 0) use block range from environment (or just the latest block if not provided) to fetch for latest events.
+    // Else, if running in loop mode (pollingDelay != 0), start with the latest block and on next loops continue from where the last one ended.
```

Comment:
good comment. 
nit: line length should not exceed 120 chars.

Reply:
I got lazy by relying on linter and forgot it does not cover comments :)